packages/p2p/src/pages/advertiser-page/__tests__/advertiser-page.spec.tsx

```diff
@@ -0,0 +1,270 @@
+            name: 'name',
+        },
+        is_counterparty_advertiser_blocked: false,
+        onMount: jest.fn(),
+        onTabChange: jest.fn(),
+        setIsDropdownMenuVisible: jest.fn(),
+        onUnmount: jest.fn(),
+        onCancel: jest.fn(),
+        is_loading: false,
+    },
+    general_store: {
+        advertiser_id: 'id2',
+        advertiser_info: {
+            name: 'my name',
+        },
+        block_unblock_user_error: '',
+        error_code: '',
+        active_index: 0,
+        setBlockUnblockUserError: jest.fn(),
+        setActiveIndex: jest.fn(),
+        path: {
+            my_profile: 3,
+        },
+        is_block_unblock_user_loading: false,
+    },
+    buy_sell_store: {
+        show_advertiser_page: true,
+        hideAdvertiserPage: jest.fn(),
+    },
+    my_profile_store: {
+        setActiveTab: jest.fn(),
+    },
+};
+
+jest.mock('../advertiser-page-profile', () => jest.fn(() => <div>profile</div>));
+jest.mock('../advertiser-page-adverts', () => jest.fn(() => <div>adverts</div>));
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    Loading: jest.fn(() => <div> loading...</div>),
+}));
+
+jest.mock('Stores', () => ({
+    ...jest.requireActual('Stores'),
+    useStores: jest.fn(() => mock_store),
+}));
+
+jest.mock('react-router-dom', () => ({
+    ...jest.requireActual('react-router-dom'),
+    useHistory: () => ({
+        push: jest.fn(),
+    }),
+}));
+
+describe('<AdvertisePage/>', () => {
+    it('should render advertiser page in default state', () => {
+        render(<AdvertisePage />);
+
+        expect(screen.getByText("Advertiser's page")).toBeInTheDocument();
+        expect(screen.getByText('adverts')).toBeInTheDocument();
+    });
+    it('should render advertiser page with overlay when advertiser is blocked', () => {
+        (useStores as jest.Mock).mockReturnValue({
+            ...mock_store,
+            advertiser_page_store: {
+                ...mock_store.advertiser_page_store,
+                is_counterparty_advertiser_blocked: true,
+            },
+        });
+        render(<AdvertisePage />);
+
+        expect(screen.getByText('You have blocked test name.')).toBeInTheDocument();
+    });
+    it('should open BlockUserModal when unblock button is clicked', () => {
+        (useStores as jest.Mock).mockReturnValue({
+            ...mock_store,
+            advertiser_page_store: {
+                ...mock_store.advertiser_page_store,
+                is_counterparty_advertiser_blocked: true,
+            },
+        });
+        render(<AdvertisePage />);
+
+        userEvent.click(screen.getByRole('button', { name: 'Unblock' }));
+
+        expect(mock_modal_manager.showModal).toHaveBeenCalledWith({ key: 'BlockUserModal' });
+    });
+    it('should close advertiser page when page back button is clicked and set counterparties as active tab when navigated from profile page to advertiser page', () => {
```

Comment:
do you think we can reduce this title to the one below?

```suggestion
    it('Should close advertiser page and activate counterparties tab on back button click', () => {
```